File: app/src/main/java/dev/ragnarok/fenrir/model/MarketAlbum.java
package dev.ragnarok.fenrir.model;

import android.os.Parcel;

public class MarketAlbum extends AbsModel {

    public static final Creator<MarketAlbum> CREATOR = new Creator<MarketAlbum>() {
        @Override
        public MarketAlbum createFromParcel(Parcel in) {
            return new MarketAlbum(in);
        }

        @Override
        public MarketAlbum[] newArray(int size) {
            return new MarketAlbum[size];
        }
    };

    private final int id;
    private final int owner_id;
    private String access_key;
    private String title;
    private Photo photo;
    private int count;
    private int updated_time;

    public MarketAlbum(int id, int owner_id) {
        this.id = id;
        this.owner_id = owner_id;
    }

    protected MarketAlbum(Parcel in) {
        super(in);
        id = in.readInt();
        owner_id = in.readInt();
        access_key = in.readString();
        count = in.readInt();
        updated_time = in.readInt();
        title = in.readString();
        photo = in.readParcelable(Photo.class.getClassLoader());
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeInt(id);
        dest.writeInt(owner_id);
        dest.writeString(access_key);
        dest.writeInt(count);
        dest.writeInt(updated_time);
        dest.writeString(title);
        dest.writeParcelable(photo, flags);
    }

    public int getId() {
        return id;
    }

    public int getOwner_id() {
        return owner_id;
    }

    public String getAccess_key() {
        return access_key;
    }

    public MarketAlbum setAccess_key(String access_key) {
        this.access_key = access_key;
        return this;
    }

    public String getTitle() {
        return title;
    }

    public MarketAlbum setTitle(String title) {
        this.title = title;
        return this;
    }

    public Photo getPhoto() {
        return photo;
    }

    public MarketAlbum setPhoto(Photo photo) {
        this.photo = photo;
        return this;
    }

    public int getCount() {
        return count;
    }

    public MarketAlbum setCount(int count) {
        this.count = count;
        return this;
    }

    public int getUpdated_time() {
        return updated_time;
    }

    public MarketAlbum setUpdated_time(int updated_time) {
        this.updated_time = updated_time;
        return this;
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/MarketAlbum.kt
package dev.ragnarok.fenrir.model

import android.os.Parcel
import android.os.Parcelable
import dev.ragnarok.fenrir.readTypedObjectCompat
import dev.ragnarok.fenrir.writeTypedObjectCompat

class MarketAlbum : AbsModel {
    private val id: Int
    private val owner_id: Int
    private var access_key: String? = null
    private var title: String? = null
    private var photo: Photo? = null
    private var count = 0
    private var updated_time: Long = 0

    constructor(id: Int, owner_id: Int) {
        this.id = id
        this.owner_id = owner_id
    }

    internal constructor(`in`: Parcel) : super(`in`) {
        id = `in`.readInt()
        owner_id = `in`.readInt()
        access_key = `in`.readString()
        count = `in`.readInt()
        updated_time = `in`.readLong()
        title = `in`.readString()
        photo = `in`.readTypedObjectCompat(Photo.CREATOR)
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        super.writeToParcel(parcel, i)
        parcel.writeInt(id)
        parcel.writeInt(owner_id)
        parcel.writeString(access_key)
        parcel.writeInt(count)
        parcel.writeLong(updated_time)
        parcel.writeString(title)
        parcel.writeTypedObjectCompat(photo, i)
    }

    fun getId(): Int {
        return id
    }

    fun getOwner_id(): Int {
        return owner_id
    }

    fun getAccess_key(): String? {
        return access_key
    }

    fun setAccess_key(access_key: String?): MarketAlbum {
        this.access_key = access_key
        return this
    }

    fun getTitle(): String? {
        return title
    }

    fun setTitle(title: String?): MarketAlbum {
        this.title = title
        return this
    }

    fun getPhoto(): Photo? {
        return photo
    }

    fun setPhoto(photo: Photo?): MarketAlbum {
        this.photo = photo
        return this
    }

    fun getCount(): Int {
        return count
    }

    fun setCount(count: Int): MarketAlbum {
        this.count = count
        return this
    }

    fun getUpdated_time(): Long {
        return updated_time
    }

    fun setUpdated_time(updated_time: Long): MarketAlbum {
        this.updated_time = updated_time
        return this
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MarketAlbum> {
        override fun createFromParcel(parcel: Parcel): MarketAlbum {
            return MarketAlbum(parcel)
        }

        override fun newArray(size: Int): Array<MarketAlbum?> {
            return arrayOfNulls(size)
        }
    }
}